app/pages/corrections/VatPeriodCorrectionsListPage.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.corrections
+
+import controllers.actions.AuthenticatedControllerComponents
+import controllers.routes
+import models.corrections.PeriodWithCorrections
+import models.{Index, Period, UserAnswers}
+import pages.{CheckYourAnswersPage, Page, Waypoints}
+import play.api.mvc.Call
+import queries.{AllCorrectionPeriodsQuery, CorrectionPeriodQuery, DeriveNumberOfCorrectionPeriods}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Success, Try}
+
+case class VatPeriodCorrectionsListPage(period: Period, addAnother: Boolean) extends Page {
+ override def nextPageNormalMode(waypoints: Waypoints, answers: UserAnswers): Page = {
+        if(addAnother) {
+          answers.get(DeriveNumberOfCorrectionPeriods) match {
+            case Some(size) =>
+              ???
+            //Todo: Add when VEIOSS-371 merged
```

Comment:
this has now been merged, so you should be able to implement this

Reply:
Done. Thanks